Remove dns

bowei opened the pull request:
This PR removes the DNS server code from the main kubernetes repo. All of the relevant patches have been applied to kubernetes/dns (e.g. the commit history).

bowei committed:
Update image references to the output of the kubernetes/dns project

bowei committed:
Remove DNS code from the main repo (code is now in http://github.com/kubernetes/dns)

bowei committed:
Remove kube-dns from golang.sh

bowei committed:
Result of hack/update-all.sh

k8s-reviewable commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.kubernetes.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/39448)
<!-- Reviewable:end -->

bowei commented:
@k8s-bot test this

bowei commented:
@k8s-bot test this

bowei commented:
@k8s-bot test this

rmmh commented:
@k8s-bot test this

bowei commented:
@k8s-bot test this

bowei commented:
@k8s-bot test this

bowei commented:
@k8s-bot test this

bowei commented:
@k8s-bot test this

rmmh commented:
@k8s-bot test this

rmmh commented:
@k8s-bot test this

rmmh commented:
@k8s-bot test this

bowei commented:
@k8s-bot test this

bowei commented:
@thockin @MrHohn

thockin commented on test/e2e/dns_configmap.go:
```diff
@@ -43,6 +42,7 @@ type dnsConfigMapTest struct {
 	labels []string
 
 	cm      *v1.ConfigMap
+	fedMap  map[string]string
```
what is this e2e change doing in this commit?

bowei replied:
split into separate commit

thockin commented on cluster/addons/dns/kubedns-controller.yaml.in:
```diff
@@ -124,7 +124,7 @@ spec:
             cpu: 150m
             memory: 10Mi
       - name: sidecar
-        image: gcr.io/google_containers/k8s-dns-sidecar-amd64:1.10.0
```
merge with first commit?

bowei replied:
merged

thockin commented:
I don't know why the docs are now adding that footer.  will have to figure out.  You should strike those changes, I think.

MrHohn commented:
Would be better to also link the new DNS repo in https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/dns/README.md.

rmmh commented:
@k8s-bot test this

bowei committed:
Update DNS readme to point to the new code repository

bowei committed:
Remove reference to dns code from e2e test

This reference was not necessary and avoids having to vendor dns
back into Kubernetes for now.

bowei commented:
@MrHohn -- added reference to new repository
@thockin -- removed the docs changes

MrHohn commented:
LGTM

thockin commented:
Make sure any docs on the main docs repo are also updated

eparis commented:
Uggh. Wish there was any reasonable way to detect when things like this are going to break the godeps verification :-(

bowei commented:
@eparis  -- which verification broke?

thockin commented:
I'm guessing that there are now vendor'ed pkgs that are not needed?  Is
there a hueristic we can use to trigger that test?

On Mon, Jan 9, 2017 at 10:33 AM, Bowei Du <notifications@github.com> wrote:

> @eparis <https://github.com/eparis> -- which verification broke?
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/39448#issuecomment-271365427>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFVgVPgF6-OJCnt30fAfef0__eZ30_yeks5rQn2TgaJpZM4LbJs7>
> .
>

eparis commented:
verify-godeps.  We only run that check by default when someone touches Godeps/ or vendor/ because it take a long time and hack/verfiy-all.sh takes long enough.

In this case you removed code, but you didn't touch Godeps. So the tests for this PR shortcutted the test.

When https://k8s-testgrid.appspot.com/google-unit#verify-master run however it does not bypass and always runs the full godep verification.

Your code removal makes some of the files in Godep/vendor no longer needed. So now the verification fails.

I just wish there was some fast way to tell that we don't need a package in `vendor` anymore

eparis commented:
I'm not sure how to speed things up until a solution to https://github.com/golang/go/issues/13078 comes along. It just takes a long time to clone that much stuff.

thockin commented:
Some variant of:

```
find . -type f -name \*.go | xargs -n1 dirname | xargs go list -f '{{range
.Deps}}{{.}}{{"\n"}}{{end}}' | sort | uniq > /tmp/all-kube-godeps
find vendor/ -type d | sed 's/^vendor\///' | while read V; do grep -q "$V"
/tmp/all-kube-godeps || echo "$V is unused"; done
```

...might work.  There are some false positives still, but you're a shell
wizard, I am sure you can figure it out :)  Need to consider binaries we
build and vendor that are not literally linked from kube code...

On Mon, Jan 9, 2017 at 10:44 AM, Eric Paris <notifications@github.com>
wrote:

> I'm not sure how to speed things up until a solution to golang/go#13078
> <https://github.com/golang/go/issues/13078> comes along. It just takes a
> long time to clone that much stuff.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/39448#issuecomment-271368460>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFVgVOHRg-uSaTlHEjzARG5QV5klaLwsks5rQoAlgaJpZM4LbJs7>
> .
>

eparis commented:
Last time I played with it I remembed it being a particular PITA because I needed to run it for multiple GOOS settings. Maybe I'll keep playing.

thockin commented:
Here's what that shows me against  head:

```
github.com/go-openapi/validate is unused
github.com/inconshreveable is unused
github.com/inconshreveable/mousetrap is unused
github.com/ugorji/go/codec/codecgen is unused
github.com/seccomp is unused
github.com/seccomp/libseccomp-golang is unused
github.com/rackspace/gophercloud/testhelper/client is unused
github.com/jteeuwen/go-bindata/go-bindata is unused
github.com/Microsoft is unused
github.com/Microsoft/go-winio is unused
github.com/Azure/go-ansiterm is unused
github.com/Azure/go-ansiterm/winterm is unused
github.com/docker/docker/pkg/term/windows is unused
github.com/docker/docker/pkg/longpath is unused
github.com/stretchr/testify/require is unused
```

so yeah, GOOS=Windows will be needed

On Mon, Jan 9, 2017 at 11:14 AM, Eric Paris <notifications@github.com>
wrote:

> Last time I played with it I remembed it being a particular PITA because I
> needed to run it for multiple GOOS settings. Maybe I'll keep playing.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/39448#issuecomment-271378676>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFVgVBd9ALdw-ow1EktWR9U7J-S-2dH1ks5rQocLgaJpZM4LbJs7>
> .
>
